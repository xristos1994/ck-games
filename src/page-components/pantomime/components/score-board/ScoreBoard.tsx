import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { teams } from '@models/pantomime/props';
import { IState } from '@models/interfaces';
// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require('./ScoreBoard.module.css');

interface IProps {
  teams: {
    id: number;
    score: number;
    name: string;
  }[];
  t: ITranslate;
}

export const _ScoreBoard: FC<IProps> = ({ teams, t }): ReactElement => {
  const sortedTeams = [...teams]
    .sort((t1, t2) => (t1.score < t2.score ? 1 : t1.score > t2.score ? -1 : 0))
    .map(({ id, score, name }) => ({
      id,
      score,
      name
    }));

  return (
    <div className={classnames(styles.scoreBoardContainer)}>
      <div className={classnames(styles.scoreBoardTitle)}>{t('Scoreboard')}</div>
      <div className={classnames(styles.scoreBoard)}>
        {sortedTeams.map((team, index) => [
          <div key={`index_${team.id}`}>{index + 1}.</div>,
          <div key={`name_${team.id}`}>{team.name}</div>,
          <div key={`score_${team.id}`}>{team.score}</div>
        ])}
      </div>
    </div>
  );
};

const ScoreBoard = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        teams: IProps['teams'];
      },
      {
        teams: IProps['teams'];
      }
    >({
      teams
    })
  ),
  withTranslation
)(_ScoreBoard);

export { ScoreBoard };
