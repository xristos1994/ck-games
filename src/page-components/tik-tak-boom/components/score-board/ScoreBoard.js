import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { classnames } from "@utils/component-utils";
import { players } from "@models/tik-tak-boom/props";
import styles from "./styles.module.css";

const _ScoreBoard = ({ players }) => {
  return (
    <div className={styles.scoreBoardContainer}>
      <table className={styles.scoreBoard}>
        <caption className={styles.scoreBoardTitle}>Score board</caption>
        <tbody>
          {[...players]
            .sort((p1, p2) =>
              p1.numOfBooms < p2.numOfBooms
                ? 1
                : p1.numOfBooms > p2.numOfBooms
                ? -1
                : 0
            )

            .map((player, index) => (
              <tr
                key={player.id}
                className={classnames(styles.playerRow, {
                  [styles.inactive]: !player.isActive,
                })}
              >
                <td key="index" className={styles.indexCell}>
                  {index + 1}.
                </td>
                <td key="name" className={styles.nameCell}>
                  {player.name}
                </td>
                <td key="score" className={styles.scoreCell}>
                  {player.numOfBooms}
                </td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

const ScoreBoard = connect(
  createStructuredSelector({
    players,
  })
)(_ScoreBoard);

export { ScoreBoard };
