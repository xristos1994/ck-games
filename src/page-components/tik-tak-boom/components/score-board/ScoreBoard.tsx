import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { players } from '@models/tik-tak-boom/props';
import { IState } from '@models/interfaces';
// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require('./ScoreBoard.module.css');

interface IProps {
  players: {
    id: number;
    isActive: boolean;
    numOfBooms: number;
    name: string;
  }[];
  t: ITranslate;
}

export const _ScoreBoard: FC<IProps> = ({ players, t }): ReactElement => {
  return (
    <div className={classnames(styles.scoreBoardContainer)}>
      <div className={classnames(styles.scoreBoardTitle)}>{t('Scoreboard')}</div>
      <div className={classnames(styles.scoreBoard)}>
        {players.map((player, index) => [
          <div
            key={`index_${player.id}`}
            className={classnames({
              [styles.inactive]: !player.isActive
            })}
          >
            {index + 1}.
          </div>,
          <div
            key={`name_${player.id}`}
            className={classnames({
              [styles.inactive]: !player.isActive
            })}
          >
            {player.name}
          </div>,
          <div
            key={`score_${player.id}`}
            className={classnames({
              [styles.inactive]: !player.isActive
            })}
          >
            {player.numOfBooms}
          </div>
        ])}
      </div>
    </div>
  );
};

const ScoreBoard = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        players: IProps['players'];
      },
      {
        players: IProps['players'];
      }
    >({
      players: (state: IState): IProps['players'] =>
        [...players(state)]
          .sort((p1, p2) => (p1.numOfBooms < p2.numOfBooms ? 1 : p1.numOfBooms > p2.numOfBooms ? -1 : 0))
          .map(({ id, isActive, numOfBooms, name }) => ({
            id,
            isActive,
            numOfBooms,
            name
          }))
    })
  ),
  withTranslation
)(_ScoreBoard);

export { ScoreBoard };
