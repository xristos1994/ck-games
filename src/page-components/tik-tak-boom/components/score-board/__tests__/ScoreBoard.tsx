import React from 'react';
import renderer from 'react-test-renderer';
import { _ScoreBoard as ScoreBoard } from './../ScoreBoard';

const t = (label: string, param?: string[]) => label + (param ? param[0] : '');

describe('page-components/TikTakBoom/ScoreBoard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ScoreBoard
        players={[
          { id: 1, isActive: true, numOfBooms: 3, name: 'Player 1' },
          { id: 1, isActive: false, numOfBooms: 0, name: 'Player 2' },
          { id: 1, isActive: true, numOfBooms: 1, name: 'Player 3' },
          { id: 1, isActive: false, numOfBooms: 4, name: 'Player 4' }
        ]}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
