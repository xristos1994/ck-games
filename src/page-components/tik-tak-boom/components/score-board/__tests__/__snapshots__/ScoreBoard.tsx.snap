// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/TikTakBoom/ScoreBoard renders correctly 1`] = `
<div
  className="scoreBoardContainer"
>
  <div
    className="scoreBoardTitle"
  >
    Scoreboard
  </div>
  <div
    className="scoreBoard"
  >
    <div
      className=""
    >
      1
      .
    </div>
    <div
      className=""
    >
      Player 1
    </div>
    <div
      className=""
    >
      3
    </div>
    <div
      className="inactive"
    >
      2
      .
    </div>
    <div
      className="inactive"
    >
      Player 2
    </div>
    <div
      className="inactive"
    >
      0
    </div>
    <div
      className=""
    >
      3
      .
    </div>
    <div
      className=""
    >
      Player 3
    </div>
    <div
      className=""
    >
      1
    </div>
    <div
      className="inactive"
    >
      4
      .
    </div>
    <div
      className="inactive"
    >
      Player 4
    </div>
    <div
      className="inactive"
    >
      4
    </div>
  </div>
</div>
`;
